core/src/main/java/org/elasticsearch/http/netty/NettyHttpChannel.java

```diff
@@ -58,19 +58,22 @@
     private final NettyHttpServerTransport transport;
     private final Channel channel;
     private final org.jboss.netty.handler.codec.http.HttpRequest nettyRequest;
-    private OrderedUpstreamMessageEvent orderedUpstreamMessageEvent = null;
+    private final OrderedUpstreamMessageEvent orderedUpstreamMessageEvent;
+    private final long reservedBytes;
 
     public NettyHttpChannel(NettyHttpServerTransport transport, NettyHttpRequest request,
```

Comment:
can we remove this constructor variant and document that you can pass null for orderedUpstreamMessageEvent ?